public/JavaScript/gallery.js

```diff
@@ -0,0 +1,33 @@
+document.addEventListener('DOMContentLoaded', () => {
+  let selectedAnimalType = null
+
+  // Function to hide all gallery items
+  const hideAllAnimals = (elements) => {
+    elements.forEach(item => {
+      item.style.display = 'none'
+    })
+  }
+  // Function to display selected animal
+  const showSelectedAnimals = () => {
+    hideAllAnimals(document.querySelectorAll('.gallery-items'))
+    if (selectedAnimalType) {
+      document.querySelectorAll(`.gallery-items[data-animal-type="${selectedAnimalType}"]`).forEach(item => {
```

Comment:
you already looked for and stored all the gallery-items. You don't need to go and search all of them again